phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -571,6 +573,8 @@ public static PName newPName(byte[] keyBuffer, int keyOffset, int keyLength) {
     // before 4.15, so that we can rollback the upgrade to 4.15 if required
     private boolean allowSplittableSystemCatalogRollback;
 
+    private MetricsMetadataSource metricsSource;
```

Comment:
Should this also be declared volatile ? I noticed you declared another similar variable volatile below.

Reply:
Is one necessary? In the other cases I was populating the variable in a synchronized block with a null check, which I believe needs a volatile, but in this case I'm just setting it from start() and not treating it as an explicit singleton. (Do I need to?)

Reply:
I see. It is not needed then.